cfgov/jinja2/v1/_includes/organisms/video-player.html

```diff
@@ -1,23 +1,132 @@
 {# ==========================================================================
 
-   Video Player
+   video_player.render()
 
    ==========================================================================
 
    Description:
 
-   Create a video player for the Wagtail Video Player module when given:
+   Builds Video markup when given:
 
-   value: Object defined from a StreamField block.
+   video_id:        YouTube video ID, e.g. "dQw4w9WgXcQ".
 
-   value.video_url: A string containing the YouTube embed URL to use.
+   button_pos:      String used to denote the position of the play button.
+                    Possible values: 'center (default), bottomRight'
+
+   thumbnail_url:   Optional alternate image URL to show before video is played
+                    or after it is finished.
+
+   is_fcm:          Boolean. True if this video is being rendered as part of a
+                    featured content module.
 
    ========================================================================== #}
 
-{% import 'macros/video-player.html' as player with context %}
+{% macro render(
+    video_id,
+    button_pos='center',
+    thumbnail_url=none,
+    is_fcm=false
+) -%}
+    {# This image only gets displayed if JavaScript is disabled. #}
+    {%- set image_url =
+        thumbnail_url or
+        static( 'img/cfpb_video_cover_card_1380x776.png' )
+    %}
+    {#
+        Exclude the origin parameter on preview views to avoid breaking video
+        playback when the preview request host (from the Wagtail Site) doesn't
+        match the URL serving the video.
+    #}
+    {% set video_url =
+        'https://www.youtube.com/embed/' ~
+        video_id ~
+        '?enablejsapi=1' ~
+        (
+            ('&origin=' ~ request.scheme ~ '://' ~ request.get_host())
```

Comment:
Would `request.scheme` ever not be `https`?

Reply:
When running with localhost or in some Docker container accessed over HTTP, it might not be.